src/replay/service.zig

```diff
@@ -409,8 +435,49 @@ pub fn run(deps: ReplayDependencies) !void {
     defer zone.deinit();
 
     var state = try ReplayState.init(deps);
+    // This defer is defined first, so it runs last. This ensures that any threads
+    // borrowing `state.slot_tracker_rw`/`state.epoch_tracker_rw` are joined before the RwMuxes are deinitialized.
     defer state.deinit();
 
+    // Start vote listener inside replay so it can reference replay state directly.
+    var vote_tracker: sig.consensus.VoteTracker = .EMPTY;
+    defer vote_tracker.deinit(deps.allocator);
+
+    // TODO: Should `slot_data_provider` itself be wrapped in RwMux to ensure consistency
+    // of the struct itself across multiple threads?
```

Comment:
I don't see why that would be necessary if the fields are protected already. These pointers themselves should never be written again after this.